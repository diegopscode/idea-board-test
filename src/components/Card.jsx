import React from 'react'

const updateData = () => {

}

const Card = ({ index, title, text, autoFocus, onRemove, onChange, onBlur }) => (
    <div className="card card-idea">
        <div className="card-content">
            <div className="card-title">
                <input type="text" name="title" className="card-input" value={title} onChange={e => onChange(e.target, index)} autoFocus={autoFocus} onBlur={onBlur} />
            </div>
            <div className="card-text">
                <textarea type="text" name="text" rows="8" className="card-input" value={text} onChange={e => onChange(e.target, index)} onBlur={onBlur} />
            </div>
        </div>
        <a className="card-button-close" onClick={() => onRemove(index)}>X</a>
    </div>
)

export default Card
