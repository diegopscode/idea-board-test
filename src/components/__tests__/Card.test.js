import React from 'react'
import renderer from 'react-test-renderer'

import Card from '../Card'

describe("CardComponent", () => {
    test('snapshot renders card component', () => {
        const component = renderer.create(<Card title="Teste" text="..." />)
        let card = component.toJSON()
        expect(card).toMatchSnapshot()
    })
})