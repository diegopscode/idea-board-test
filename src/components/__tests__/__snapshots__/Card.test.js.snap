// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent snapshot renders card component 1`] = `
<div
  className="card card-idea"
>
  <div
    className="card-content"
  >
    <div
      className="card-title"
    >
      <input
        className="card-input"
        name="title"
        onChange={[Function]}
        placeholder="Enter a title"
        type="text"
        value="Teste"
      />
    </div>
    <div
      className="card-text"
    >
      <textarea
        className="card-input"
        name="text"
        onChange={[Function]}
        placeholder="Describe your idea"
        rows="8"
        type="text"
        value="..."
      />
    </div>
  </div>
  <a
    className="card-button-close"
    onClick={[Function]}
  >
    Ã—
  </a>
</div>
`;
